src/zcash/Proof.cpp

```diff
@@ -163,6 +163,10 @@ curve_G2 CompressedG2::to_libsnark_g2() const
 
     assert(r.is_well_formed());
 
+    if (alt_bn128_modulus_r * r != curve_G2::zero()) {
+        throw std::runtime_error("point is not in G2");
+    }
+
```

Comment:
This check is now included in the protocol spec (2016.0-beta-1.11): https://github.com/zcash/zips/commit/c791075ed1f3f1e6152c86528df2746efa692be8